example/tostring_jsonenums.go

```diff
@@ -0,0 +1,93 @@
+// generated by jsonenums -stringer=true -type=toString; DO NOT EDIT
+
+package main
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+)
+
+var (
+	_toStringNameToValue = map[string]toString{
+		"elemA": elemA,
+		"elemB": elemB,
+		"elemC": elemC,
+	}
+
+	_toStringValueToName = map[toString]string{
+		elemA: "elemA",
+		elemB: "elemB",
+		elemC: "elemC",
+	}
+)
+
+func init() {
+	var v toString
+	if _, ok := interface{}(v).(fmt.Stringer); ok {
+		_toStringNameToValue = map[string]toString{
+			interface{}(elemA).(fmt.Stringer).String(): elemA,
+			interface{}(elemB).(fmt.Stringer).String(): elemB,
+			interface{}(elemC).(fmt.Stringer).String(): elemC,
+		}
+	}
+}
+
+func (r toString) String() string {
+	return _toStringValueToName[r]
```

Comment:
Have we really thought through that this is how we want this to behave?

As it stands if someone makes an invalid value of the relevant type, we'll get an empty string out of this. In every other scenario we get an error.

Maybe that's ok, but I'm at least mildly concerned at the moment